SwanCustomEnvironments/swancustomenvironments/serverextension.py

```diff
@@ -0,0 +1,88 @@
+from tornado import web
+from traitlets import Unicode
+from traitlets.config import Configurable
+
+from jupyter_server.base.handlers import JupyterHandler, APIHandler
+from jupyter_server.utils import url_path_join
+
+import subprocess
+import re
+from os import path
+
+
+class SwanCustomEnvironments(Configurable):
+
+    makenv_path = Unicode(
+        path.join(path.dirname(__file__), "scripts/makenv.sh"),
+        config=True,
+        help='Path to the script used to create custom environments.'
+    )
+
+
+class SwanCustomEnvironmentsApiHandler(APIHandler):
+
+    config = None
+
+    def initialize(self):
+        self.config = SwanCustomEnvironments(config=self.config)
+
+    @web.authenticated
+    def get(self):
+        self.set_header("Content-Type", "text/event-stream")
+
+        env_name = self.get_query_argument("env", default=None)
+        requirements = self.get_query_argument("req", default=None)
```

Comment:
This would be renamed to `repository` if we follow the convention mentioned in my previous comments.